radis/pgsearch/providers.py

```diff
@@ -0,0 +1,183 @@
+from radis.search.site import Search, SearchResult
+from radis.search.utils.query_parser import BinaryNode, ParensNode, QueryNode, TermNode, UnaryNode
+
+from ..reports.models import Report
+from .utils.document_utils import document_from_pgsearch_response
+
+logger = logging.getLogger(__name__)
+labels = LANGUAGE_LABELS
+
+
+# def _build_filter_dict(filters: SearchFilters) -> list[dict]:
+#     qf = []
+
+#     f = {"term": {"groups": filters.group}}
+#     qf.append(f)
+
+#     if filters.study_date_from or filters.study_date_till:
+#         f = {"range": {"study_datetime": {}}}
+#         if filters.study_date_from:
+#             f["range"]["study_datetime"]["gte"] = filters.study_date_from
+#         if filters.study_date_till:
+#             f["range"]["study_datetime"]["lte"] = filters.study_date_till
+#         qf.append(f)
+
+#     if filters.study_description:
+#         f = {"wildcard": {"study_description": f"*{filters.study_description}*"}}
+#         qf.append(f)
+
+#     if filters.modalities:
+#         f = {"terms": {"modalities": filters.modalities}}
+#         qf.append(f)
+
+#     if filters.patient_sex:
+#         f = {"term": {"patient_sex": filters.patient_sex}}
+#         qf.append(f)
+
+#     if filters.patient_age_from or filters.patient_age_till:
+#         f = {"range": {"patient_age": {}}}
+#         if filters.patient_age_from:
+#             f["range"]["patient_age"]["gte"] = filters.patient_age_from
+#         if filters.patient_age_till:
+#             f["range"]["patient_age"]["lte"] = filters.patient_age_till
+#         qf.append(f)
+
+#     if filters.language:
+#         f = {"term": {"language": filters.language}}
+#         qf.append(f)
+
+#     return qf
+
+
+def _build_query_string(node: QueryNode) -> str:
+    if isinstance(node, TermNode):
+        if node.term_type == "WORD":
+            return node.value
+        elif node.term_type == "PHRASE":
+            return f'"{node.value.replace('"', '\\"')}"'
+        else:
+            raise ValueError(f"Unknown term type: {node.term_type}")
+    elif isinstance(node, ParensNode):
+        return f"({_build_query_string(node.expression)})"
+    elif isinstance(node, UnaryNode):
+        return f"{node.operator} {_build_query_string(node.operand)}"
+    elif isinstance(node, BinaryNode):
+        if node.implicit:
+            return f"{_build_query_string(node.left)} {_build_query_string(node.right)}"
+        return (
+            f"{_build_query_string(node.left)} {node.operator} "
+            + f"{_build_query_string(node.right)}"
+        )
+    else:
+        raise ValueError(f"Unknown node type: {type(node)}")
+
+
+def build_filter_query(search: Search):
+    query = Q()
+    query_str = _build_query_string(search.query)
+
+    # Ensure you use a valid text search configuration
+    language_config = LANGUAGE_LABELS[search.filters.language].lower()
+    search_vector = SearchVector("body", config=language_config)
```

Comment:
I think the main problem here is that the `body` is not indexed, and you already did the indexing in `update_search_vector`. There is no need to create another `SearchVector` instance (which doesn't use any index at all).